### DISCUSSION TITLE
preg_replace is faster than str_replace on PHP 7.x.x

### DISCUSSION THREAD

**Discussion by gigamaster** [xoopscube/legacy#216] [General]
Test converting all instances of http:// to https://  
It seeems that preg_replace is 2x faster on PHP 7.x.x  

```php
// start time
$start = microtime(true);

// Add a long string
$url = 'http://domain-name.com/long-string-here repeat .....';

$urlnew = '';

// Sample to reflect a bigger html output
// add a zero '0' to the value 7000
// simulate thousands of lines of code
for($i=0; $i <= 7000; $i++){
  $urlnew .= $i.$url;
}
//echo $urlnew;

$url = $urlnew;
 
// Run preg then str to get the time
$url = preg_replace('/http\:/', 'https:', $url);
//$url = str_replace('http:', 'https:', $url);

// end time 
$end = microtime(true); 
printf("<p>Time for execution %f seconds.</p>\n", $end - $start);
```

**PHP online**
- https://3v4l.org/
- https://onlinephp.io/

**Result PHP 7.x**  
preg_replace: 0.0252 seconds

str_replace: 0.0541 seconds

**Result PHP 5.6**  
preg_replace: 0.0346 seconds

str_replace: 0.0201 seconds

**Comment by nbuy** [comment]
This is include interesting things.
At first say my conclution. Use preg_* functions! 😀

I think that result reason is maintainance activity.

The str_* functions only maintenance by php community.
Also that functions old design and exist some alternate in php functions. (Low motivation)

preg_* is maintenance by php+perl community.
And that is core part of perl functions. (High motivation)

So more reviced in preg_* functions.
